app/services/uphold/refresher.rb

```diff
@@ -0,0 +1,35 @@
+# frozen_string_literal: true
+
+module Uphold
+  class Refresher
+    def self.build
+      new(impl_refresher: Uphold::Models::Authorization.new)
+    end
+
+    def initialize(impl_refresher:)
+      @impl_refresher = impl_refresher
+    end
+
+    # Makes a request to the Uphold API to refresh the current access_token
+    def call(uphold_connection:)
+      # Ensure we have an refresh_token.
+      refresh_token = uphold_connection.refresh_token
+
+      return if refresh_token.blank?
+      return if uphold_connection.authorization_expires_at&. > Time.zone.now
```

Comment:
Would you create a method called `uphold_connection.authorization_expired?`? In case we need to re-use the same method.

Reply:
Sure.

Reply:
Done.